lib/ret/hub_invite.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Ret.HubInvite do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  alias Ret.{Hub, HubInvite, Repo}
+
+  @schema_prefix "ret0"
+  @primary_key {:hub_invite_id, :id, autogenerate: true}
+
+  schema "hub_invites" do
+    field(:hub_invite_sid, :string)
+    field(:state, HubInvite.State)
+    belongs_to(:hub, Hub, references: :hub_id)
+    timestamps()
+  end
+
+  def invite_for_hub(%Hub{} = hub) do
```

Comment:
Might not be clear from the name that this can result in an insert. Maybe `get_or_create_invite_for_hub/1`?